jiant/models.py

```diff
@@ -946,8 +968,17 @@ def _seq_gen_forward(self, batch, task, predict):
         sent, sent_mask = self.sent_encoder(batch["inputs"], task)
         out["n_exs"] = get_batch_size(batch)
 
+        if isinstance(task, (CharSeq2SeqTask)):
+            decoder = getattr(self, "%s_decoder" % task.name)
+            out.update(decoder.forward(sent, sent_mask, batch["targs"]))
+            task.scorer1(out["loss"].item())
+
         if "targs" in batch:
-            pass
+            # logits: batch_size * seq_len * tgt_voc_size
+            target = batch["targs"]["words"][:, 1:].contiguous()
```

Comment:
is `[1:]` because you want to avoid <SOS> tags ? Shouldn't those be removed from the target in preprocessing (shifting by 1)?